agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -1642,7 +1642,9 @@ func TestFirelensWithS3ConfigFluentd(t *testing.T) {
 		Key:    aws.String("testfiles/fluentd.conf"),
 		Body:   strings.NewReader(content),
 	})
-	require.NoError(t, err, "unable to upload config file to s3 which is needed for the test")
+	if err != nil {
+		t.Skipf("Skipping the test since unable to upload config file to s3: %v", err)
+	}
```

Comment:
Is it possible to check permission before attemping to upload?

Failure to upload will always mean that the permission is missing? Could it be an indicator of something else going wrong, can we check the error message and based on that skip the test?

Reply:
yeah, updated to only skip test on access denied